### PR TITLE
drop tax_rate from Product and ProductType

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want to drop unused tax_rate from Product and ProductType

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/product/types/products.py [hunk 1]
```diff
@@ -832,13 +832,6 @@ def resolve_tax_type(root: models.ProductType, info):
         tax_data = info.context.plugins.get_tax_code_from_object_meta(root)
         return TaxType(tax_code=tax_data.code, description=tax_data.description)
 
-    @staticmethod
-    def resolve_tax_rate(root: models.ProductType, _info, **_kwargs):
```
Comment (maarcingebala) [inline saleor/graphql/product/types/products.py:836]: We drop the resolver, but the field is still available in the type, is that what we want, or we should remove the field at all?

Comment (d-wysocki) [reply saleor/graphql/product/types/products.py:836]: We should remove the field.

Comment (maarcingebala) [reply saleor/graphql/product/types/products.py:836]: But this field is a public API (no staff permission required) and it wasn't deprecated before. We should deprecate it first in Saleor 2.11. We also need to mention it in the changelog as a breaking change.

**Review by fowczarek — APPROVED** [review]

**Comment by maarcingebala** [comment]
Please also update the changelog.

**Review by korycins — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**